src/brevitas/core/bit_width/parameter.py

```diff
@@ -88,11 +89,21 @@ def forward(self) -> Tensor:
 
     def _load_from_state_dict(self, state_dict, prefix, local_metadata, strict,
                               missing_keys, unexpected_keys, error_msgs):
+        bit_width_const_key = prefix + 'bit_width'
         bit_width_offset_key = prefix + 'bit_width_offset'
+        if bit_width_const_key in state_dict:
+            assert bit_width_offset_key not in state_dict, "both should not be true"
+            state_dict = OrderedDict(
```

Comment:
I don't think we need to rewrite the whole dict here, we can do just two accesses rather than N:
```python
state_dict[bit_width_offset_key] = v - self.bit_width_base
del state_dict[bit_width_const_key]
```